pythonPackages.distributed: 1.15.1 -> 1.22.1

costrouc opened the pull request:
###### Motivation for this change 

Wanted to run dask distributed on python 2.7 and 3+. Additionally wanted all tests to pass. A lot of work had to be put into adding testing frameworks and ensuring that all the underlying libraries were also compatible.

###### Things done

For all the following all tests pass and run on python 2.7, 3+ now.

pythonPackages.distributed: 1.15.1 -> 1.22.1 and refactor
pythonPackages.pytest-repeat: init at 0.6.0
pythonPackages.joblib: 0.21.1 -> 0.21.2 and refactor
pythonPackages.python-lz4: init at 2.1.0
pythonPackages.pytest-faulthandler: init at 1.5.0
pythonPackages.pytest-timeout: refactor

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

costrouc commented:
Requires myself to be added to maintainers (pull request #44655) list before this pull request

xeji commented:
@GrahamcOfBorg eval

dotlambda commented on pkgs/development/python-modules/distributed/default.nix:
```diff
@@ -0,0 +1,52 @@
+
+  # py.test not picking up local config file, even when running
+  # manually: E ValueError: no option named '--runslow'
+  doCheck = false;
```
Maybe you need to add pytest-repeat or pytest-faulthandler to checkInputs.

dotlambda replied:
See https://github.com/dask/distributed/blob/master/continuous_integration/travis/install.sh#L57

costrouc committed:
pythonPackages.pytest-timeout: refactor add python 2.7 compatibiliy

In version 1.3.1 of `pytest-timeout` (latest) the setup.py file is not
compatible with python 2.7 and 3+ due to how `open` is called. A patch
has been added that can be removed once a fix and new relase is given.

costrouc committed:
pythonPackages.pytest-faulthandler: init at 1.5.0

 - compatible with 2.7 and 3+
 - all test pass.

costrouc committed:
pythonPackages.python-lz4: init at 2.1.0

Fetching from github repository instead of pypi so that all tests can
be run.

 - compatible with 2.7, 3+
 - all tests pass

costrouc committed:
pythonPackages.joblib: 0.12.1 -> 0.12.2

In order to get all the tests passing the Github repository was
downloaded instead of from pypi so that the `conftest.py` is
available.

In addition to updating the version:

  - compatible with 2.7, 3+ now
  - all tests are running and passing

costrouc committed:
pythonPackages.pytest-repeat: init at 0.6.0

 - compatible with 2.7, 3+
 - all tests pass and running

costrouc committed:
pythonPackages.distributed: 1.15.1 -> 1.22.1

The github repository was downloaded instead of the pypi repository
for testing (needed `conftest.py`). Major work was done on the
underlying dependencies to make distributed work on python 2.7,
3+. Note that the test **do** take a significant amount of time (10-15
minutes).

 - moved to `python-modules`
 - compatible with 2.7, 3+
 - all tests pass (previously tests were not run)

costrouc commented:
@dotlambda I would really like to thank you for how detailed you have been in your code reviews. I have learned a lot on how to properly package with nix and several git techniques I didn't know how to do (squash, reset --soft, etc.) really helped me with learning how to create good looking commits.

There are a lot of packages updated in the following commit. It was necessary in order to make all the tests run and ensure that it also worked on python 2.7. Looks like it was neglected in some of the packages that I have refactored.

Some questions I have about style.
  - is the goal to eventually move all packages out of `pkgs/top-level/python-packages.nix` -> `pkgs/development/python-modules/...`?
  - pypi archives most of the time do have all the tests and setup. Is it good form to use the github repository instead? At least here we can run all the tests

dotlambda commented:
> is the goal to eventually move all packages out of pkgs/top-level/python-packages.nix -> pkgs/development/python-modules/...?

Yes, and I think more than 2/3 have already been moved.

> pypi archives most of the time do not have all the tests and setup. Is it good form to use the github repository instead? At least here we can run all the tests

Fetching from GitHub has the disadvantage that our batch upgrade script won't work until https://github.com/NixOS/nixpkgs/pull/41149 is done. I sometimes open upstream PRs to add tests to the tarball, e.g. https://github.com/ethereum/eth-hash/pull/20.

dotlambda commented on pkgs/development/python-modules/pytest-repeat/default.nix:
```diff
@@ -0,0 +1,28 @@
+  '';
+
+  meta = {
+    description = "pytest plugin for repeating tests";
```
Capitalize: https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

dotlambda commented on pkgs/development/python-modules/distributed/default.nix:
```diff
@@ -0,0 +1,59 @@
+  # ignore 5 cli tests out of 1000 total tests that fail due to subprocesses
+  # these tests are not critical to the library (only the cli)
+  checkPhase = ''
+    py.test distributed -m "not avoid-travis" -r s --timeout-method=thread --timeout=300 --durations=20 --ignore="distributed/cli/tests"
```
Do you know if there's a way to set the timeout to infinity?

costrouc replied:
Not sure. I have been following the test instructions that is used on travis.

https://github.com/dask/distributed/blob/master/continuous_integration/travis/run_tests.sh#L1

dotlambda replied:
From https://bitbucket.org/pytest-dev/pytest-timeout/:

> Setting a timeout to 0 seconds disables the timeout, so if you have a global timeout set you can still disable the timeout by using the mark.

However, I'd prefer to just get rid of `pytest-timeout` if that's possible. Otherwise, just set `--timeout=0`.

dotlambda replied:
Btw you should also try leaving out the other options if possible, especially `--durations` sounds like something we'd want to avoid. Hydra is usually under heavy load, so tests will run slower than usual.

dotlambda commented:
@GrahamcOfBorg build python2.pkgs.distributed python3.pkgs.distributed

dotlambda commented:
@costrouc These commits all look very nice. Thank you for your thorough work!

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=70d9ba0e-023a-4fc2-ad9b-48f137e8f30f)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/clzis8sxxbjxrgcpzj59jd1zpqpwdmjs-python3.6-joblib-0.12.2.drv': 1 dependencies couldn't be built
test_pytest_timeout.py ...................................               [100%]

========================== 35 passed in 94.51 seconds ==========================
cannot build derivation '/nix/store/ixjalzamq475gcf8nkm6lqwymaijfdb0-python2.7-distributed-1.22.1.drv': 3 dependencies couldn't be built
test_pytest_timeout.py ...................................               [100%]

========================= 35 passed in 121.42 seconds ==========================
cannot build derivation '/nix/store/6vbqnpivvxj2vjx2ylq633i6lkpb79hn-python3.6-distributed-1.22.1.drv': 3 dependencies couldn't be built
error: build of '/nix/store/6vbqnpivvxj2vjx2ylq633i6lkpb79hn-python3.6-distributed-1.22.1.drv', '/nix/store/ixjalzamq475gcf8nkm6lqwymaijfdb0-python2.7-distributed-1.22.1.drv' failed
```
</p></details>

costrouc commented on pkgs/development/python-modules/pytest-timeout/default.nix:
```diff
@@ -9,10 +9,16 @@ buildPythonPackage rec {
+  # remove patch after version 1.3.1
+  # assuming they fix installation on python 2.7 and 3.6
+  # they have encoding issue
+  patches = [ ./0001-open-encoding.patch ];
```
There is an issue with their commit as now it does not compile for python 2.7 :man_facepalming: . So they tried to fix a 3+ issue and created a 2.7 issue. 

This is the important line that they need

```
from codec import open
```

dotlambda replied:
So, `import io` and `io.open()` don't work?

costrouc replied:
Turns out that I was not including `io` in `open` yes that patch works I will add this.

dotlambda replied:
A big advantage of using `fetchpatch` is that it saves disk space on every checkout/download of Nixpkgs, i.e. on every computer of a Nix user.

dotlambda commented on pkgs/development/python-modules/python-lz4/default.nix:
```diff
@@ -0,0 +1,43 @@
+    sha256 = "1vjfplj37jcw1mf8l810dv76dx0raia3ylgyfy7sfsb3g17brjq6";
+  };
+
+  buildInputs = [ setuptools_scm pkgconfig ];
```
I think `pytestrunner` is missing here.

dotlambda replied:
Maybe you didn't encounter `distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('pytest-runner')` because you have sandboxing turned off? You can turn it on with `nix build --option sandbox true` I think.

costrouc replied:
Thank you for showing `--option sandbox true` yes I can reproduce the issue.

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=9bcea24e-33c5-418d-abe5-a5bedb2ceefe)

Attempted: python2.pkgs.pytest-timeout, python3.pkgs.pytest-timeout, python2.pkgs.pytest-faulthandler, python3.pkgs.pytest-faulthandler

<details><summary>Partial log (click to expand)</summary><p>

```
plugins: timeout-1.3.1
collected 35 items

test_pytest_timeout.py ...................................               [100%]

========================= 35 passed in 517.85 seconds ==========================
test_pytest_timeout.py ...................................               [100%]

========================= 35 passed in 527.26 seconds ==========================
error: build of '/nix/store/n77309chczgsd6bbvkl6cjj4q5v6jw2q-python3.6-pytest-faulthandler-1.5.0.drv', '/nix/store/x1sdwl4z2mfhl301ha5ifnrr89if7cj8-python2.7-pytest-faulthandler-1.5.0.drv' failed
```
</p></details>

dotlambda commented on pkgs/development/python-modules/python-lz4/default.nix:
```diff
@@ -0,0 +1,43 @@
+
+  checkPhase = ''
+    py.test
+  '';
```
If you add `pytestrunner` above, this can be removed.

GrahamcOfBorg commented:
Unexpected error: command failed with exit code 1 on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=67e4f6d8-bd19-48e8-bb53-59939dfd8481)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/rplfrxvynrzr52fvlh84n2alflm0n9i8-python2.7-pytest-timeout-1.3.1/lib
patching script interpreter paths in /nix/store/rplfrxvynrzr52fvlh84n2alflm0n9i8-python2.7-pytest-timeout-1.3.1
running install tests
============================= test session starts ==============================
platform darwin -- Python 2.7.15, pytest-3.6.3, py-1.5.4, pluggy-0.6.0
rootdir: /private/tmp/nix-build-python2.7-pytest-timeout-1.3.1.drv-0/pytest-timeout-1.3.1, inifile: tox.ini
plugins: timeout-1.3.1
collected 35 items

[31;1merror:[0m unexpected end-of-file
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=cb0a9a97-d2e8-4e84-967d-003a567115a0)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed

<details><summary>Partial log (click to expand)</summary><p>

```
                                        ^~
building of '/nix/store/0bsfv0mr6vf1xvi11nm6fd52adjhz3aq-python2.7-pandas-0.23.3.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/d6yshyck6a178j2sfdswq3zy8xr86b1p-python2.7-partd-0.3.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/mx6yr91v3v9kvs60h7sb4izj01a5qjzg-python2.7-dask-0.18.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/9mzxjjdyzwsxxq3sdh47vc111b2n889d-python2.7-distributed-1.22.1.drv': 4 dependencies couldn't be built
building of '/nix/store/qvrv5d2dw6zxdmqxc311fq0f4ixwvdpm-python3.6-pandas-0.23.3.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/22hdlj49g7a3zjsnh2n8sfz85wqsrbng-python3.6-partd-0.3.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/c3mvsf26s2sb639p3mvxdrjx5c2ycjsc-python3.6-dask-0.18.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/cdg226yxavzz9zyp9j7d5533p41rby9s-python3.6-distributed-1.22.1.drv': 4 dependencies couldn't be built
error: build of '/nix/store/9mzxjjdyzwsxxq3sdh47vc111b2n889d-python2.7-distributed-1.22.1.drv', '/nix/store/cdg226yxavzz9zyp9j7d5533p41rby9s-python3.6-distributed-1.22.1.drv' failed
```
</p></details>

costrouc commented:
I have made all of the requested changes.

The one change I have not made is 

> Do you know if there's a way to set the timeout to infinity?

This is because I am just copying the tests that were run on travis ci.

Otherwise all done and thanks for the help!

Ekleog commented:
(triage) From what I understand, this is waiting for:
 * Removal of the lines pointed to by https://github.com/NixOS/nixpkgs/pull/44657/files#r208727864
 * At least setting `timeout` to `0` as per https://github.com/NixOS/nixpkgs/pull/44657/files#r208964330

@costrouc, it's your move :+1: Thank you for contributing!

costrouc commented:
@Ekleog I have made the additional changes thanks for pointing them out. I have tested that with the changes all packages build with `--option sandbox true`.

costrouc commented:
Should be ready for merge now with all the requested changes made.

dotlambda commented on pkgs/development/python-modules/pytest-faulthandler/default.nix:
```diff
@@ -0,0 +1,34 @@
+
+  buildInputs = [ setuptools_scm ];
+  checkInputs = [ pytest-mock ];
+  propagatedBuildInputs = [ pytest ] ++ lib.optional (pythonOlder "3.0") faulthandler;
```
Don't propagate pytest, it's a special case. Plugins should only use it for build time of itself.

dotlambda commented on pkgs/development/python-modules/pytest-faulthandler/default.nix:
```diff
@@ -0,0 +1,34 @@
+  '';
+
+  meta = {
+    description = "py.test plugin that activates the fault handler module for tests";
```
Capitalize.

dotlambda commented on pkgs/development/python-modules/python-lz4/default.nix:
```diff
@@ -0,0 +1,40 @@
+
+  buildInputs = [ setuptools_scm pkgconfig pytestrunner ];
+  checkInputs = [ pytest psutil ];
+  propagatedBuildInputs = [ ] ++ lib.optionals (!isPy3k) [ future ];
```
Drop `[ ] ++`.

dotlambda commented on pkgs/development/python-modules/joblib/default.nix:
```diff
@@ -1,29 +1,35 @@
   };
 
-  checkInputs = [ sphinx numpydoc pytest ];
+  checkInputs = [ sphinx numpydoc pytest python-lz4 ];
```
I think we can include `python-lz4` in `propagatedBuildInputs`. It's their "faster alternative to zlib and gzip for compressed serialization."

dotlambda commented on pkgs/development/python-modules/pytest-repeat/default.nix:
```diff
@@ -0,0 +1,30 @@
+  };
+
+  buildInputs = [ setuptools_scm ];
+  propagatedBuildInputs = [ pytest ];
```
Don't propagate pytest, it's a special case. Plugins should only use it for build time of itself.

costrouc committed:
Merge branch 'master' into costrouc/dask-distributed

ryantm reviewed (changes requested):
Wow, looks like this was a lot of hard work! Please fix the merge conflict.

costrouc committed:
merge: fixing changes with nixpkgs since pull-request

 - pytest-timeout: no longer requires patch
 - joblib: was improperly merged with the github merge tool (never
   using again)

costrouc commented:
@ryantm ready for merge. Thanks for the reminder!

costrouc commented:
@ryantm I will never use the github merge editor for a quick fix. Thats why the builds failed. I have fixed the issue and tested that the builds succeed now for both joblib and distributed.

ryantm commented:
@GrahamcOfBorg build python2.pkgs.distributed python3.pkgs.distributed python2.pkgs.joblib python3.pkgs.joblib

GrahamcOfBorg commented:
<!--REQUEST_ID=c86362f4-4cf9-4071-9443-0b8add242df4-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=cdb330b0-bac2-4ccb-9cc6-750f18521b24)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed, python2.pkgs.joblib, python3.pkgs.joblib

<details><summary>Partial log (click to expand)</summary><p>

```
3.04s call     distributed/deploy/tests/test_adaptive.py::test_worker_keys
3.00s call     distributed/deploy/tests/test_local.py::test_silent_startup
2.98s call     distributed/tests/test_client_executor.py::test_cancellation
2.63s call     distributed/tests/test_profile.py::test_basic
2.51s call     distributed/tests/test_failed_workers.py::test_restart_sync_no_center
2.37s call     distributed/tests/test_failed_workers.py::test_restart_sync
2.36s call     distributed/diagnostics/tests/test_progress.py::test_AllProgress
== 1 failed, 1051 passed, 114 skipped, 5 xfailed, 4 xpassed in 434.46 seconds ==
builder for '/nix/store/84m9s23yl37v3kwl8xzcks03x74sc6ix-python3.6-distributed-1.22.1.drv' failed with exit code 1
error: build of '/nix/store/84m9s23yl37v3kwl8xzcks03x74sc6ix-python3.6-distributed-1.22.1.drv', '/nix/store/fmmqhlivdvpqzif035d70awqqilqnfm1-python2.7-distributed-1.22.1.drv', '/nix/store/iwxnw0vy2g76f9h1qrvkfcyrwpsh60mf-python2.7-joblib-0.12.4.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=c86362f4-4cf9-4071-9443-0b8add242df4-->
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=2df7148d-1c74-4042-8262-7f36ba6f100f)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed, python2.pkgs.joblib, python3.pkgs.joblib

<details><summary>Partial log (click to expand)</summary><p>

```
distributed/tests/test_as_completed.py ..........                        [ 21%]
distributed/tests/test_asyncio.py ................s..                    [ 22%]
distributed/tests/test_asyncprocess.py ..........                        [ 23%]
distributed/tests/test_batched.py .......s.s..                           [ 24%]
distributed/tests/test_client.py .......................s.........ss.... [ 28%]
........................sss........X.ss....s.....s.......s.............. [ 34%]
................ssss.............................................s...... [ 40%]
............................/nix/store/i6vl5lwlz5jbkg4r6p340dwmj6fha3xq-stdenv-linux/setup: line 1290:  3123 Aborted                 py.test distributed -m "not avoid-travis" -r s --timeout-method=thread --timeout=0 --durations=20 --ignore="distributed/cli/tests"
builder for '/nix/store/h486hzbaqscqa8acbm0qaryjknlf073d-python3.6-distributed-1.22.1.drv' failed with exit code 134
error: build of '/nix/store/h486hzbaqscqa8acbm0qaryjknlf073d-python3.6-distributed-1.22.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=c86362f4-4cf9-4071-9443-0b8add242df4-->
Timed out, unknown build status on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=12cd3a0d-2b9f-477c-a912-9726a69a5317)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed, python2.pkgs.joblib, python3.pkgs.joblib

<details><summary>Partial log (click to expand)</summary><p>

```
../../nix/store/8j80sfr6rmn7yd161v1f0f1zqm28b1g6-python3.6-pandas-0.23.4/lib/python3.6/site-packages/pandas/tests/test_strings.py . [  4%]
building of '/nix/store/b5j1ydbjjxkn2nvnxsyc4rsllrzlbkm5-python2.7-pandas-0.23.4.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/ygqffarb34clqzghzfilrzjniv6viw95-python2.7-partd-0.3.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/6vr0l3c25khg7s4wj5lwpm4n2jan7sxc-python2.7-dask-0.18.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/kf7c7bsaxzckjf14m7k57h86vfi0flgl-python2.7-distributed-1.22.1.drv': 2 dependencies couldn't be built
building of '/nix/store/qa3qf1ckgx2aiq3ngqm1ir2dqqf3wzp1-python3.6-pandas-0.23.4.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/qjrb7bqk05ylddl5kz7qzalhl5jmlz3y-python3.6-partd-0.3.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/niacdp5h3hpc0zclqcz7mcqj8wlmamls-python3.6-dask-0.18.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/9gaq1bb2x16nfrpfc37lc3542zr1kca7-python3.6-distributed-1.22.1.drv': 2 dependencies couldn't be built
error: build of '/nix/store/9gaq1bb2x16nfrpfc37lc3542zr1kca7-python3.6-distributed-1.22.1.drv', '/nix/store/9w6xk081d3r5cs8gkchn4gc56f94nfga-python3.6-joblib-0.12.4.drv', '/nix/store/gws5ach6mbyyjkh44pqssssrf8wfwsd4-python2.7-joblib-0.12.4.drv', '/nix/store/kf7c7bsaxzckjf14m7k57h86vfi0flgl-python2.7-distributed-1.22.1.drv' failed
```
</p></details>

costrouc committed:
fixup! merge: fixing changes with nixpkgs since pull-request

costrouc commented:
I have been trying to reproduce the failed tests and everytime I ignore a test another magically seems to fail. I think that turning off the tests is fine. There are 800 tests and only 1 or 2 seem to fail each time. @ryantm is this acceptable? I am sorry that this has taken so long.

ryantm commented:
@costrouc There are a lot of python packages with the tests disabled. Obviously it is preferred to have them enabled, but if they can't be made reliable, it is better to not have them. Hopefully the package maintainer runs their own tests as part of the release process.

ryantm reviewed (changes requested):
please fix test reliability

costrouc commented:
Fixed. I set `doCheck = false` for `pythonPackages.distributed`

ryantm commented:
@GrahamcOfBorg build python2.pkgs.distributed python3.pkgs.distributed python2.pkgs.joblib python3.pkgs.joblib

GrahamcOfBorg commented:
<!--REQUEST_ID=75dddcbc-6f26-411c-bfb8-8a220523a792-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=37c3440e-21a9-4173-ac5a-86fe9e26ba40)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed, python2.pkgs.joblib, python3.pkgs.joblib

<details><summary>Partial log (click to expand)</summary><p>

```
[DEBUG:MainProcess:QueueManagerThread] closing call_queue
[DEBUG:MainProcess:QueueManagerThread] telling queue thread to quit
[DEBUG:MainProcess:QueueManagerThread] joining processes
[DEBUG:MainProcess:QueueFeederThread] feeder thread got sentinel -- exiting
[DEBUG:MainProcess:QueueManagerThread] queue management thread clean shutdown of worker processes: []
[DEBUG:MainProcess:MainThread] running the remaining "atexit" finalizers
/nix/store/2q33qrmnphzc7hr4pdams7cb8z6sbqm7-python2.7-distributed-1.22.1
/nix/store/93ipmi0zapd8ckfxrrgrqsld7jdbl62i-python3.6-distributed-1.22.1
/nix/store/24i4w686088i6k6pmxkrd2k8cnfgiil9-python2.7-joblib-0.12.4
/nix/store/13ldlwmd3qj4ihhkn0cp2g0agb1klxkc-python3.6-joblib-0.12.4
```
</p></details>

ryantm commented:
Woo! 🎉

GrahamcOfBorg commented:
<!--REQUEST_ID=75dddcbc-6f26-411c-bfb8-8a220523a792-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=9d2a3aa1-a548-4537-b457-2f2f81f9038a)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed, python2.pkgs.joblib, python3.pkgs.joblib

<details><summary>Partial log (click to expand)</summary><p>

```
[DEBUG:MainProcess:Dummy-677] Interpreter shutting down. Waking up queue_manager_threads [(<Thread(QueueManagerThread, started daemon 123145423290368)>, <joblib.externals.loky.process_executor._ThreadWakeup instance at 0x10b84c4d0>)]
[DEBUG:MainProcess:QueueManagerThread] queue management thread shutting down
[DEBUG:MainProcess:QueueManagerThread] closing call_queue
[DEBUG:MainProcess:QueueManagerThread] telling queue thread to quit
[DEBUG:MainProcess:QueueManagerThread] joining processes
[DEBUG:MainProcess:QueueFeederThread] feeder thread got sentinel -- exiting
[DEBUG:MainProcess:QueueManagerThread] queue management thread clean shutdown of worker processes: []
[DEBUG:MainProcess:Dummy-677] running the remaining "atexit" finalizers
builder for '/nix/store/iwxnw0vy2g76f9h1qrvkfcyrwpsh60mf-python2.7-joblib-0.12.4.drv' failed with exit code 1
error: build of '/nix/store/iwxnw0vy2g76f9h1qrvkfcyrwpsh60mf-python2.7-joblib-0.12.4.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=75dddcbc-6f26-411c-bfb8-8a220523a792-->
Timed out, unknown build status on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.44657&attempt_id=a853938c-7dde-4652-a2b5-d11c9cea15d3)

Attempted: python2.pkgs.distributed, python3.pkgs.distributed, python2.pkgs.joblib, python3.pkgs.joblib

<details><summary>Partial log (click to expand)</summary><p>

```
building of '/nix/store/9w6xk081d3r5cs8gkchn4gc56f94nfga-python3.6-joblib-0.12.4.drv' timed out after 3600 seconds
building of '/nix/store/b5j1ydbjjxkn2nvnxsyc4rsllrzlbkm5-python2.7-pandas-0.23.4.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/ygqffarb34clqzghzfilrzjniv6viw95-python2.7-partd-0.3.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/6vr0l3c25khg7s4wj5lwpm4n2jan7sxc-python2.7-dask-0.18.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/bfc1k8ncmfqij286lkm75zi63vv4zdyw-python2.7-distributed-1.22.1.drv': 1 dependencies couldn't be built
building of '/nix/store/qa3qf1ckgx2aiq3ngqm1ir2dqqf3wzp1-python3.6-pandas-0.23.4.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/qjrb7bqk05ylddl5kz7qzalhl5jmlz3y-python3.6-partd-0.3.8.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/niacdp5h3hpc0zclqcz7mcqj8wlmamls-python3.6-dask-0.18.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/fv7f8y18rhl0zcy0zqmacbrv5aawmb0a-python3.6-distributed-1.22.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/9w6xk081d3r5cs8gkchn4gc56f94nfga-python3.6-joblib-0.12.4.drv', '/nix/store/bfc1k8ncmfqij286lkm75zi63vv4zdyw-python2.7-distributed-1.22.1.drv', '/nix/store/fv7f8y18rhl0zcy0zqmacbrv5aawmb0a-python3.6-distributed-1.22.1.drv', '/nix/store/gws5ach6mbyyjkh44pqssssrf8wfwsd4-python2.7-joblib-0.12.4.drv' failed
```
</p></details>

ryantm commented:
@costrouc It looks like the tests of `joblib` are failing on darwin :(
